spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -560,6 +578,17 @@ protected void doInit() {
 				setContainerProperties(container);
 				container.afterPropertiesSet();
 				this.replyContainer = container;
+				if (this.correlationKey == null) {
+					logger.warn("'async=true' requires a correlationKey; ignored");
```

Comment:
???
No guaranty that I have set `async=true` before.
Looks like condition has been missed around this `warn`.